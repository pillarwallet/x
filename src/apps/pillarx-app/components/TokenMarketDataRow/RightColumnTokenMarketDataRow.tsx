import { TbTriangleFilled } from 'react-icons/tb';

// types
import { TokensMarketDataRow } from '../../../../types/api';

// utils
import { limitDigitsNumber } from '../../../../utils/number';

// components
import HighDecimalsFormatted from '../HighDecimalsFormatted/HighDecimalsFormatted';
import BodySmall from '../Typography/BodySmall';

type RightColumnTokenMarketDataRowProps = {
  data: TokensMarketDataRow;
};

const RightColumnTokenMarketDataRow = ({
  data,
}: RightColumnTokenMarketDataRowProps) => {
  const { rightColumn } = data;

  return (
    <div className="flex flex-col h-full items-end justify-between flex-shrink-0">
      <div className="flex desktop:gap-1 tablet:gap-1 mobile:flex-col mobile:items-end">
        {rightColumn?.line1?.price ? (
          <HighDecimalsFormatted
            value={limitDigitsNumber(Number(rightColumn.line1.price || 0))}
            moneySymbol="$"
            styleNumber={`font-normal ${rightColumn?.line1?.direction === 'UP' && 'desktop:text-market_row_green tablet:text-market_row_green'} ${rightColumn?.line1?.direction === 'DOWN' && 'desktop:text-percentage_red tablet:text-percentage_red'} mobile:text-white desktop:text-base tablet:text-base mobile:text-sm`}
            styleZeros="mobile:text-white desktop:text-xs tablet:text-xs mobile:text-[10px]"
          />
        ) : null}
        <div
          className={`flex gap-1 items-center desktop:px-1 desktop:rounded tablet:px-1 tablet:rounded ${rightColumn?.line1?.direction === 'UP' && 'text-market_row_green desktop:bg-market_row_green/[.1] tablet:bg-market_row_green/[.1] mobile:bg-transparent'} ${rightColumn?.line1?.direction === 'DOWN' && 'text-percentage_red desktop:bg-percentage_red/[.1] tablet:bg-percentage_red/[.1] mobile:bg-transparent'}`}
        >
          {rightColumn?.line1?.direction === 'UP' ||
          rightColumn?.line1?.direction === 'DOWN' ? (
            <TbTriangleFilled
              size={6}
              color={
                rightColumn?.line1?.direction === 'UP' ? '#5CFF93' : '#FF366C'
              }
              style={{
                transform:
                  rightColumn?.line1?.direction === 'DOWN'
                    ? 'rotate(180deg)'
                    : 'none',
              }}
            />
          ) : null}
          {rightColumn?.line1?.percentage ? (
            <BodySmall className="font-normal">
              {rightColumn?.line1?.percentage}
            </BodySmall>
          ) : null}
        </div>
      </div>
      {rightColumn?.line2?.transactionCount ? (
        <BodySmall className="font-normal mobile:hidden text-white">
          <span className="text-white/[.5]">Txs:</span>{' '}
          {rightColumn?.line2?.transactionCount}
        </BodySmall>
      ) : null}
    </div>
  );
};

export default RightColumnTokenMarketDataRow;
