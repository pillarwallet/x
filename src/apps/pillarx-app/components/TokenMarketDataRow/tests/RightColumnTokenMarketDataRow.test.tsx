import { render, screen } from '@testing-library/react';

// components
import RightColumnTokenMarketDataRow from '../RightColumnTokenMarketDataRow';

const ethTokenRow = {
  link: 'token-atlas?someLink=true',
  leftColumn: {
    token: {
      primaryImage: 'eth.png',
    },
    line1: {
      text1: 'ETH',
      text2: 'Ethereum',
      copyLink: '0xD76b5c2A23ef78368d8E34288B5b65D616B746aE',
    },
    line2: {
      timestamp: 1745331519,
      volume: '1.2m',
      liquidity: '30,123',
    },
  },
  rightColumn: {
    line1: {
      price: '0.042188',
      direction: 'UP',
      percentage: '20.1%',
    },
    line2: {
      transactionCount: '1823',
    },
  },
};

describe('<RightColumnTokenMarketDataRow /> - ETH token row', () => {
  it('renders and matches snapshot', () => {
    const tree = render(<RightColumnTokenMarketDataRow data={ethTokenRow} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders percentage, price and transaction count ', () => {
    render(<RightColumnTokenMarketDataRow data={ethTokenRow} />);

    expect(screen.getByText('$0.042188')).toBeInTheDocument();
    expect(screen.getByText('20.1%')).toBeInTheDocument();
    expect(screen.getByText(/Txs:/)).toBeInTheDocument();
    expect(screen.getByText('1823')).toBeInTheDocument();
  });

  it('renders with missing percentage', () => {
    const noPercentageRow = {
      ...ethTokenRow,
      rightColumn: {
        line1: {
          price: '0.999',
          direction: 'UP',
        },
      },
    };

    render(<RightColumnTokenMarketDataRow data={noPercentageRow} />);
    expect(screen.getByText('$0.999')).toBeInTheDocument();
    expect(screen.queryByText('%')).not.toBeInTheDocument();
  });

  it('renders only transaction count when line1 is missing', () => {
    const noLine1Row = {
      ...ethTokenRow,
      rightColumn: {
        line2: {
          transactionCount: '9999',
        },
      },
    };

    render(<RightColumnTokenMarketDataRow data={noLine1Row} />);
    expect(screen.getByText('9999')).toBeInTheDocument();
    expect(screen.queryByText('$')).not.toBeInTheDocument();
  });
});
