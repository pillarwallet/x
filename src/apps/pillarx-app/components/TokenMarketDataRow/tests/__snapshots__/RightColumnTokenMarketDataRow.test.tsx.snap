// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RightColumnTokenMarketDataRow /> - ETH token row renders and matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col h-full items-end justify-between flex-shrink-0"
      >
        <div
          class="flex desktop:gap-1 tablet:gap-1 mobile:flex-col mobile:items-end"
        >
          <p
            class="font-normal desktop:text-market_row_green tablet:text-market_row_green false mobile:text-white desktop:text-base tablet:text-base mobile:text-sm"
          >
            $
            0.04219
          </p>
          <div
            class="flex gap-1 items-center desktop:px-1 desktop:rounded tablet:px-1 tablet:rounded text-market_row_green desktop:bg-market_row_green/[.1] tablet:bg-market_row_green/[.1] mobile:bg-transparent false"
          >
            <svg
              color="#5CFF93"
              fill="none"
              height="6"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              style="color: rgb(92, 255, 147); transform: none;"
              viewBox="0 0 24 24"
              width="6"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M12 1.67a2.914 2.914 0 0 0 -2.492 1.403l-8.11 13.537a2.914 2.914 0 0 0 2.484 4.385h16.225a2.914 2.914 0 0 0 2.503 -4.371l-8.116 -13.546a2.917 2.917 0 0 0 -2.494 -1.408z"
                fill="currentColor"
                stroke-width="0"
              />
            </svg>
            <p
              class="text-sm font-medium font-normal"
            >
              20.1%
            </p>
          </div>
        </div>
        <p
          class="text-sm font-medium font-normal mobile:hidden text-white"
        >
          <span
            class="text-white/[.5]"
          >
            Txs:
          </span>
           
          1823
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col h-full items-end justify-between flex-shrink-0"
    >
      <div
        class="flex desktop:gap-1 tablet:gap-1 mobile:flex-col mobile:items-end"
      >
        <p
          class="font-normal desktop:text-market_row_green tablet:text-market_row_green false mobile:text-white desktop:text-base tablet:text-base mobile:text-sm"
        >
          $
          0.04219
        </p>
        <div
          class="flex gap-1 items-center desktop:px-1 desktop:rounded tablet:px-1 tablet:rounded text-market_row_green desktop:bg-market_row_green/[.1] tablet:bg-market_row_green/[.1] mobile:bg-transparent false"
        >
          <svg
            color="#5CFF93"
            fill="none"
            height="6"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            style="color: rgb(92, 255, 147); transform: none;"
            viewBox="0 0 24 24"
            width="6"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M12 1.67a2.914 2.914 0 0 0 -2.492 1.403l-8.11 13.537a2.914 2.914 0 0 0 2.484 4.385h16.225a2.914 2.914 0 0 0 2.503 -4.371l-8.116 -13.546a2.917 2.917 0 0 0 -2.494 -1.408z"
              fill="currentColor"
              stroke-width="0"
            />
          </svg>
          <p
            class="text-sm font-medium font-normal"
          >
            20.1%
          </p>
        </div>
      </div>
      <p
        class="text-sm font-medium font-normal mobile:hidden text-white"
      >
        <span
          class="text-white/[.5]"
        >
          Txs:
        </span>
         
        1823
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
